src/nbl/asset/builtin/cache/ICacheKeyCreator.h

```diff
@@ -0,0 +1,26 @@
+#ifndef __I_CACHE_KEY_CREATOR_H_INCLUDED__
+#define __I_CACHE_KEY_CREATOR_H_INCLUDED__
+
+#include "nbl/asset/IAsset.h"
+
+namespace nbl
+{
+	namespace asset
+	{
+		class ICacheKeyCreator
+		{
+			public:
+				virtual std::string to_string() = 0;
+
+				template<typename Type>
+				_NBL_STATIC_INLINE std::string getHexString(const Type& value)
+				{
+					std::stringstream stream;
+					stream << std::setfill('0') << std::hex << value;
```

Comment:
there's a big problem here.. without `std::setw` before every `<<value` you wont get any leading fill characters which will lead to ambiguous paths.

I.e. is a path `aaa` two variables `aa` and `a` or `aa` and `a` OR is it three variables of `a`

so you either need an illegal file path character such as `?` as a separator... or `std::setw(sizeof(T)*2ull)` before every `<<`